# Retina ganglion cells dataset
Maciej Daniszewski, Anne Senabouth, Duncan E. Crombie, Quan Nguyen, Samuel Lukowski, Tejal Kulkani,  Donald J Zack,  Alice PÃ©bay, Joseph E. Powell and Alex W. Hewitt

Public repository for code associated with the processing of single cell retina ganglion cells, from the publication [Single Cell RNA Sequencing of stem cell-derived retinal ganglion cells.](https://www.biorxiv.org/content/early/2017/09/22/191395)

## Raw Data
[E-MTAB-6108 - Single cell RNA sequencing of stem cell-derived retinal ganglion cells](https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-6108/files/)

## Required installations
### Cell Ranger 1.3.1
Please refer to the [10x Genomics](https://support.10xgenomics.com/single-cell-gene-expression/software/downloads/1.3/) for information about this program.

### R 3.4.2
Please refer to the [R website](https://www.r-project.org/) for information about this program.

## Instructions
### Processing raw data with Cell Ranger 1.3.1
**THIS STAGE SHOULD BE RUN ON A HIGH-PERFORMANCE COMPUTING ENVIRONMENT**
1. Download the raw data from ArrayExpress.
2. Download the Cell Ranger 1.3.1 reference. You can also prepare your own by following the instructions [here](https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/advanced/references).
3. Edit [project.xml](project.xml) to suit your computing environment. Make sure your paths do not end in '/'.
4. Run [ProcessData.bash](ProcessData.bash). If you are working on a cluster, you can call this from a cluster submission script. We have included one for PBSPro ([SubmitPBS.pbs](SubmitPBS.pbs)), which you can use as a template for other systems.

### Processing Cell Ranger output with R
**THIS STAGE CAN BE RUN ON A DESKTOP**
Run [ProcessData.R](ProcessData.R), where the second argument points to the location of the files generated by Cell Ranger.

```bash
Rscript ProcessData.R RetinaGanglionCells
```

This will generate two clean expression matrices in the directory.

## Exploring this dataset with 'ascend'
You can explore the clean data available on ArrayExpress by loading the expression matrices as follows:

```
# Load the expression matrices into R
sample1.matrix <- read.csv("iPSC_RGscRNASeq_Sample1.tsv", sep = "\t")
sample2.matrix <- read.csv("iPSC_RGscRNASeq_Sample2.tsv", sep = "\t")

# Connect the two matrices together using dplyr and set the rownames
library(dplyr)
expression.matrix <- full_join(sample1.matrix, sample2.matrix, by = "X")
rownames(expression.matrix) <- expression.matrix[ ,1]
expression.matrix <- expression.matrix[ ,2:ncol(expression.matrix)]

# Create an EMSet for use with ascend
em.set <- NewEMSet(ExpressionMatrix = expression.matrix)

```
